src/sid/initial_conditions.py

```diff
@@ -450,22 +451,27 @@ def _sample_factorized_virus_strains_for_infections(
 def _integrate_immune_individuals(
     states: pd.DataFrame, initial_immunity: pd.Series
 ) -> pd.DataFrame:
-    """Integrate immune individuals in states.
+    """Integrate immunity level of individuals in states.
+
+
+    TODO: The case where initial immunity is due to vaccination is  # noqa: T000
+    ignored here. This does not seem right?
 
     Args:
         states (pandas.DataFrame): The states which already include sampled infections.
-        initial_immunity (pandas.Series): A series with sampled immune individuals.
+        initial_immunity (pandas.Series): A series with sampled immunity levels of
+            individuals.
 
     Returns:
-        states (pandas.DataFrame): The states with additional immune individuals.
+        states (pandas.DataFrame): The states with initial immunity integrated.
 
     """
-    extra_immune = initial_immunity & ~states["immune"]
-    states.loc[extra_immune, "immune"] = True
-    states.loc[extra_immune, "ever_infected"] = True
-    states.loc[extra_immune, "cd_ever_infected"] = 0
-    states.loc[extra_immune, "cd_immune_false"] = states.loc[
-        extra_immune, "cd_immune_false_draws"
-    ]
+    immunity = np.maximum(initial_immunity, states["immunity"])
+    states["immunity"] = immunity
+
+    locs = immunity > 0
+    states.loc[locs, "ever_infected"] = True
+    states.loc[locs, "cd_ever_infected"] = 0
```

Comment:
Why was / is this line here? The `cd_ever_infected` should only be set to 0 when someone is infected, IMO. 

This line / its predecessor breaks the `how_to_use_initial_conditions` notebook.

Reply:
- to get the people who are assumed to have been infected before the burn in period use `initial_immunity > states["immunity"]`
- set the `cd_ever_infected` to `- (number_of_burn_in_periods + 1)`